src/helpers.h

```diff
@@ -5,14 +5,14 @@
 #include "datatypes.h"
 
 // errors if cond passed is true
-// #define ON_ERROR_EXIT(cond, message) \ 
-// do{ \
-//   if(cond){ \ 
-//     printf("ERROR in function: %s at line %d\n", __func__, __LINE__ ); \
-//     perror((message)); \ 
-//     exit(1); \ 
-//   }\
-// }while(0)
+#define ON_ERROR_EXIT(cond, message) \ 
+do{ \
+  if(cond){ \ 
+    printf("ERROR in function: %s at line %d\n", __func__, __LINE__ ); \
+    perror((message)); \ 
+    exit(1); \ 
+  }\
+}while(0)
```

Comment:
Is building working for you with the macro defined this way? I can check on my machine in the morning but it cmake was unhappy with me the last time I had the macro defined

Reply:
We can switch to using a func instead